### PR TITLE
Docs - Collision and pickup logic system

### PR DESCRIPTION
**Pull request by The0mikkel**
This pull request, should be the base for the discussion about how collision mechanics, and therefore also pickup mechanics, should work in the game.

### CODE REVIEW

**Review by JfriisJ — APPROVED** [review]
Lets discuss the best action with Mads

File: docs/diagrams/SequenceDiagrams/EnemyCollidesWithPlayer.puml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Enemy -> CollisionPart : getCollisionPart()
+activate CollisionPart
+CollisionPart -> Enemy : CollisionParts
+deactivate CollisionPart
+
+Enemy -> Player : CollisionParts
+deactivate Enemy
+
+Note right of Player
+    On enemy, a damage collision part exists
```
Comment (jakob753951) [inline docs/diagrams/SequenceDiagrams/EnemyCollidesWithPlayer.puml:55]: A damage collision part exists on Enemy*

**Review by jakob753951 — CHANGES_REQUESTED** [review]
Fix pls

**Review by jakob753951 — APPROVED** [review]
Tak :)

**Pull request merged**